src/components/pages/packages/flyouts/packageNew/packageNew.js

```diff
@@ -97,21 +117,52 @@ export class PackageNew extends LinkedComponent {
   }
 
   render() {
-    const { t, isPending, error } = this.props;
-    const { type, packageFile, changesApplied } = this.state;
+    const { t,
+      isPending,
+      error,
+      configTypes,
+      configTypesError,
+      configTypesIsPending } = this.props;
+    const {
+      packageType,
+      configType,
+      packageFile,
+      changesApplied } = this.state;
 
     const summaryCount = 1;
-    const typeOptions = packageTypeOptions.map(value => ({
-      label: t(`packages.typeOptions.${value.toLowerCase()}`),
+    const packageOptions = packageTypeOptions.map(value => ({
+      label: getPackageTypeTranslation(value, t),
       value
-    }));
+    }))
+    const configOptions = configTypes ?
+      configTypes.map(value => ({
+        label: getConfigTypeTranslation(value, t),
+        value
+      }))
+      : {};
 
     const completedSuccessfully = changesApplied && !error && !isPending;
     // Validators
     const requiredValidator = (new Validator()).check(Validator.notEmpty, t('packages.flyouts.new.validation.required'));
 
     // Links
-    this.packageTypeLink = this.linkTo('type').map(({ value }) => value).withValidator(requiredValidator);
+    this.packageTypeLink = this.linkTo('packageType').map(({ value }) => value).withValidator(requiredValidator);
+    this.configTypeLink = this.linkTo('configType')
+      .map(({ value }) => value)
+      .check(
+        // Validate for non-empty value if packageType is of type 'Device Configuration'
+        configValue => this.packageTypeLink.value === packageTypeOptions[1] ? Validator.notEmpty(configValue) : true,
```

Comment:
Referring to these with indexes is just asking for trouble. Use named constants.

Reply:
done